src/path.cr

```diff
@@ -0,0 +1,941 @@
+# A `Path` represents a filesystem path and allows path-handling operations
+# such as querying it's components as well as semantic manipulations.
+#
+# A path is hierarchical and composed of a sequence of directory and file name
+# elements separated by a special separator or delimiter. A root component,
+# that identifies a file system hierarchy, may also be present.
+# The name element that is farthest from the root of the directory hierarchy is
+# the name of a file or directory. The other name elements are directory names.
+# A Path can represent a root, a root and a sequence of names, or simply one or
+# more name elements.
+# A Path is considered to be an empty path if it consists solely of one name
+# element that is empty. Accessing a file using an empty path is equivalent
+# to accessing the default directory of the process.
+#
+# # Examples
+#
+# ```
+# Path.posix("foo/bar/baz.cr").parent   # => Path.posix("foo/bar")
+# Path.posix("foo/bar/baz.cr").basename # => "baz.cr"
+# Path.posix("./foo/../bar").normalize  # => Path.posix("bar")
+# Path.posix("~/bin").expand            # => Path.posix("/home/crystal/bin")
+# ```
+#
+# For now, its methods are purely lexical, there is no direct filesystem access.
+#
+# Path handling comes in different flavours depending on operating system:
+#
+# * `Path.posix()` creates a new POSIX path
+# * `Path.windows()` create a new Windows path
+# * `Path.native()` is an alias for `Path.posix` on POSIX platforms and
+#   `Path.windows()` on Windows platforms.
+#
+# ```
+# # On POSIX system:
+# Path.new("foo", "bar", "baz.cr") # => Path.windows("foo/bar/baz.cr")
+# # On Windows system:
+# Path.new("foo", "bar", "baz.cr") # => Path.windows("foo\\bar\\baz.cr")
+# ```
+#
+# The main differences between Windows and POSIX paths:
+# * POSIX paths use forward slash (`/`) as only path separator, Windows paths use
+#   backslash (`\`) as default separator but also recognize forward slashs.
+# * POSIX paths are generally case-sensitive, Windows paths case-insensitive
+#   (see `#<=>`).
+# * A POSIX path is absolute if it begins with a forward slash (`/`). A Windows path
+#   is absolute if it starts with a drive letter and root (`C:\`).
+#
+# ```
+# Path.posix("/foo/./bar").normalize   # => Path.posix("/foo/bar")
+# Path.windows("/foo/./bar").normalize # => Path.windows("\\foo\\bar")
+
+# Path.posix("/foo").absolute?   # => true
+# Path.windows("/foo").absolute? # => false
+#
+# Path.posix("foo") == Path.posix("FOO")    # => false
+# Path.windows("foo") == Path.windows("FOO") # => true
+# ```
+struct Path
+  include Comparable(Path)
+
+  class Error < Exception
+  end
+
+  enum Flavour : UInt8
```

Comment:
Is there any convention of which English to use? e.g. en-US, en-GB, en-CA

Reply:
IMO if any, `en-US` should be used since it's the most common dialect.

Reply:
Most of the stdlib seems to tend towards US english (unfortunately)

Reply:
I thought most of the things I have seen where en-US.  Seems to me this should follow that pattern so the interface is consistent.

Reply:
@straight-shoota Would you mind changing this to `Flavor` (ditto docs)?

Reply:
Sounds good, I'd go for `Kind`.

Reply:
This is kind of a `Platform` or `System`

Reply:
`Kind` is good.